framework/rbac/DbManager.php

```diff
@@ -885,8 +921,8 @@ public function loadFromCache()
         }
 
         $data = $this->cache->get($this->cacheKey);
-        if (is_array($data) && isset($data[0], $data[1], $data[2])) {
-            list ($this->items, $this->rules, $this->parents) = $data;
+        if (is_array($data) && isset($data[0], $data[1], $data[2], $data[3])) {
```

Comment:
This will fail if caching assignments is disabled because the fourth item is null. Should `array_keys_exists` or just `count()` be used here? Or should the `assignments` property have an empty array as the default value?